src/nvim/log.c

```diff
@@ -171,6 +162,16 @@ bool logmsg(int log_level, const char *context, const char *func_name, int line_
 #endif
 
   log_lock();
+  if (recursive) {
+    if (!did_msg) {
+      did_msg = true;
+      char *arg1 = func_name ? xstrdup(func_name) : (context ? xstrdup(context) : NULL);
+      loop_schedule_deferred(&main_loop, event_create(on_log_recursive_event, 2, arg1, line_num));
+    }
+    g_stats.log_skip++;
+    return false;
```

Comment:
does `uv_mutex_init_recursive` mean we don't need `log_unlock` here?

Reply:
oops we do.